internal/store/fake_store_test.go

```diff
@@ -839,3 +839,28 @@ func TestFakeStoreReferencePolicy(t *testing.T) {
 	assert.Nil(err)
 	assert.Len(routes, 2, "expect two ReferencePolicies")
 }
+
+func TestFakeStoreGateway(t *testing.T) {
+	assert := assert.New(t)
+
+	policies := []*gatewayv1alpha2.Gateway{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			},
+			Spec: gatewayv1alpha2.GatewaySpec{},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "bar",
+			},
+			Spec: gatewayv1alpha2.GatewaySpec{},
+		},
+	}
+	store, err := NewFakeStore(FakeObjects{Gateways: policies})
+	assert.Nil(err)
+	assert.NotNil(store)
```

Comment:
```suggestion
	require.Nil(t, err)
	require.NotNil(t, store)
```

Without this we'll continue on with `err != nil` or `store` that is `nil` (which could panic given the below invocation).

Reply:
++, require whenever the rest of the test couldn't possibly meaningfully continue.